### PR TITLE
Add workspace name as subtitle in individual modal screens

### PR DESCRIPTION
**Pull request by mollfpr**
### Details

Adding workspace name that get from props `withFullPolicy` hoc and call the `this.props.policy.name` and pass it to `HeaderWithCloseButton` component.
Update `workspace` page included:
1. Invite page - `workspace/${policyID}/invite`
2. Settings page - `workspace/${policyID}/settings`
3. Card page - `workspace/${policyID}/card`
4. Reimburse page - `workspace/${policyID}/reimburse`
5. Bills page - `workspace/${policyID}/bills`
6. Invoices page - `workspace/${policyID}/invoices`
7. Travel page - `workspace/${policyID}/travel`

### Fixed Issues

$ https://github.com/Expensify/App/issues/7840

### Tests

Open `workspace` page, open the page below, and verify that the workspace name is shown in the modal header.
1. Invite page - `workspace/${policyID}/invite`
2. Settings page - `workspace/${policyID}/settings`
3. Card page - `workspace/${policyID}/card`
4. Reimburse page - `workspace/${policyID}/reimburse`
5. Bills page - `workspace/${policyID}/bills`
6. Invoices page - `workspace/${policyID}/invoices`
7. Travel page - `workspace/${policyID}/travel`

- [x] Verify that no errors appear in the JS console

#### Contributor (PR Author) Checklist
- [x] I verified the PR has a small number of commits behind `main`
- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I clearly indicated the environment tests should be run in (Staging vs Production)
- [x] I wrote testing steps that cover success & fail scenarios (if applicable)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests & veryfy they passed on:
    - [x] iOS / native
    - [x] Android / native
    - [x] iOS / Safari
    - [x] Android / Chrome
    - [x] MacOS / Chrome
    - [x] MacOS / Desktop
- [x] I verified there are no console errors related to changes in this PR
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I added comments when the code was not self explanatory
    - [x] I put all copy / text shown in the product in all `src/languages/*` files (if applicable)
    - [x] I followed proper naming convention for platform-specific files (if applicable)
    - [x] I followed style guidelines (in [`Styling.md`](https://github.com/Expensify/App/blob/main/STYLING.md)) for all style edits I made
    - [x] I followed the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/STYLE.md#jsdocs))
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I corroborated the UI performance was not affected (the performance is the same than `main` branch)
- [x] If I created a new component I verified that a similar component doesn't exist in the codebase

#### PR Reviewer Checklist
- [ ] I verified the PR has a small number of commits behind `main`
- [ ] I verified the correct issue is linked in the `### Fixed Issues` section above
- [ ] I verified testing steps are clear and they cover the changes made in this PR
    - [ ] I verified the testing environment is mentioned in the test steps
- [ ] I verified testing steps cover success & fail scenarios (if applicable)
- [ ] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [ ] iOS / native
    - [ ] Android / native
    - [ ] iOS / Safari
    - [ ] Android / Chrome
    - [ ] MacOS / Chrome
    - [ ] MacOS / Desktop
- [ ] I verified there are no console errors related to changes in this PR
- [ ] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified comments were added when the code was not self explanatory
    - [ ] I verified any copy / text shown in the product was added in all `src/languages/*` files (if applicable)
    - [ ] I verified proper naming convention for platform-specific files was followed (if applicable)
    - [ ] I verified [style guidelines](https://github.com/Expensify/App/blob/main/STYLING.md) were followed
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/STYLE.md#jsdocs)) were followed
- [ ] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/PR_REVIEW_GUIDELINES.md)
- [ ] I verified other components are not impacted by changes in this PR (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [ ] I verified the UI performance was not affected (the performance is the same than `main` branch)
- [ ] If a new component is created I verified that a similar component doesn't exist in the codebase

### QA Steps

Open `workspace` page, open the page below, and verify that the workspace name is shown in the modal header.
1. Invite page - `workspace/${policyID}/invite`
2. Settings page - `workspace/${policyID}/settings`
3. Card page - `workspace/${policyID}/card`
4. Reimburse page - `workspace/${policyID}/reimburse`
5. Bills page - `workspace/${policyID}/bills`
6. Invoices page - `workspace/${policyID}/invoices`
7. Travel page - `workspace/${policyID}/travel`
8. Bank account page - `workspace/${policyID}/bank-account`

- [x] Verify that no errors appear in the JS console

#### Web

https://user-images.githubusercontent.com/25520267/157229371-009ad8f6-1e00-4016-aad4-b9b20594efaf.mov

#### Mobile Web

https://user-images.githubusercontent.com/25520267/157229775-9976012a-431d-4306-9c43-75f817909272.mov

#### Desktop

https://user-images.githubusercontent.com/25520267/157229898-4fd3c8bd-82cd-424e-b47a-020f7ef01aa3.mov

#### iOS

https://user-images.githubusercontent.com/25520267/157230027-b42de577-3b9c-4385-9ee6-891d56f7c570.mov

#### Android

https://user-images.githubusercontent.com/25520267/157230070-87adfa4b-188e-416b-9f2b-22cf4706e706.mov

### CODE REVIEW

**Comment by mollfpr** [comment]
re-create from https://github.com/Expensify/App/pull/8038

**Comment by rushatgabhane** [comment]
@mollfpr could you please merge `main` from upstream into your branch, thanks!

File: src/pages/workspace/WorkspaceInvitePage.js [hunk 1]
```diff
@@ -242,12 +242,14 @@ class WorkspaceInvitePage extends React.Component {
             Boolean(this.state.userToInvite),
             this.state.searchValue,
         );
+        const policyName = lodashGet(this.props.policy, 'name');
```
Comment (Luke9389) [inline src/pages/workspace/WorkspaceInvitePage.js:245]: I know this is pedantic, but could you put a new line below this to match the other ones?

**Review by rushatgabhane — CHANGES_REQUESTED** [review]
@mollfpr looks like we're in an infinite loop.
1. Go to Settings -> Workspace -> Connect bank account page
2. Go back

https://user-images.githubusercontent.com/29673073/157762705-b12b6a8d-feac-4e5b-8e54-1ff5c1fd64ef.mov

**Comment by mollfpr** [comment]
@rushatgabhane There's infinite loop render happen in hoc `withFullPolicy`. I think it's because `useNavigationState` keeps re-render.
https://github.com/Expensify/App/blob/c26e41573c6b6ea91abef85afc173ba6ecc4cd2c/src/pages/workspace/withFullPolicy.js#L89

I have 2 solutions for the problem

1. using navigation `getState()` instead of `useNavigationState` hook. This will keep the component from re-render.
```diff
- const currentRoute = _.last(useNavigationState(state => state.routes || []));
+ const currentRoute = _.last(navigationRef.current.getState().routes);
```

2. All this happens because when we try to redirect to `workspace/:policyID/bank-account` but we don't have any bank account yet so the `WorkspaceBankAccountPage` will redirect us to `/bank-account/BankAccountStep` with a function before rendering the page component. We can fix this with move the function call inside `componentDidMount` and the infinite loop will disappear.
https://github.com/Expensify/App/blob/c26e41573c6b6ea91abef85afc173ba6ecc4cd2c/src/pages/workspace/WorkspaceBankAccountPage.js#L85-L88
```js
    componentDidMount() {
        this.unsubscribe = this.props.navigation.addListener('focus', this.onScreenFocus);

        if (!this.getShouldShowPage()) {
            this.navigateToBankAccountRoute();
        }
    }
```

**Comment by rushatgabhane** [comment]
@mollfpr fantastic work finding the solutions!

The problem with solution 1 is that `path` is undefined, i.e. it doesn't re-render when we actually want it to. The cause of infinite loop still exists, it just breaks the loop because we don't re-render.

> All this happens because when we try to redirect to workspace/:policyID/bank-account but we don't have any bank account yet so the WorkspaceBankAccountPage will redirect us to /bank-account/BankAccountStep with a function before rendering the page component. We can fix this with move the function call inside componentDidMount and the infinite loop will disappear

I agree. This is the root cause, and it won't cause an infinite loop anymore. 
Let's get this done!

**Comment by mollfpr** [comment]
@rushatgabhane @Luke9389 It's all set, ready for another review.

**Review by rushatgabhane — APPROVED** [review]
@Luke9389 LGTM! 🎉

#### PR Reviewer Checklist
- [x] I verified the PR has a small number of commits behind `main`
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the testing environment is mentioned in the test steps
- [x] I verified testing steps cover success & fail scenarios (if applicable)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] iOS / native
    - [x] Android / native
    - [x] iOS / Safari
    - [x] Android / Chrome
    - [x] MacOS / Chrome
    - [x] MacOS / Desktop
- [x] I verified there are no console errors related to changes in this PR
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified comments were added when the code was not self explanatory
    - [x] I verified any copy / text shown in the product was added in all `src/languages/*` files (if applicable)
    - [x] I verified proper naming convention for platform-specific files was followed (if applicable)
    - [x] I verified [style guidelines](https://github.com/Expensify/App/blob/main/STYLING.md) were followed
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/STYLE.md#jsdocs)) were followed
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components are not impacted by changes in this PR (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified the UI performance was not affected (the performance is the same than `main` branch)
- [x] If a new component is created I verified that a similar component doesn't exist in the codebase

**Review by Luke9389 — APPROVED** [review]
Good work, thanks @mollfpr

**Pull request merged**